src/pages/tracked-contracts/TrackedContract.tsx

```diff
@@ -1,59 +1,82 @@
 import React, { useEffect, useState } from 'react';
 import { FaCheck, FaTimes, FaTrash } from 'react-icons/fa';
+import ContractsApi from '../../api/contracts';
 import Button from '../../components/common/Buttons';
 import Toggle from '../../components/common/Toggle';
-
-export interface TrackedContract {
-  address?: string;
-  enabled?: boolean;
-  edited?: boolean;
-}
+import { Contract } from '../../interfaces';
 
 interface TrackedContractProps {
-  trackedContract: TrackedContract;
-  // saved: (contract: TrackedContract) => void; // Instead do a callback as "saved to refresh the view or the array"
-  remove: () => void;
+  trackedContract: Contract;
+  // saved: (contract: TrackedContract) => void; // TODO: Once we have the BE working use this function to do a callback to the parent to update with the id or the new status
+  remove: () => void; // This helps remove the line from the array and refresh the parent view
   focused?: boolean;
 }
 
 const TrackedContractRow: React.FC<TrackedContractProps> = ({ trackedContract, remove, focused }) => {
   const [inputFocused, setInputFocused] = useState<boolean>(focused || false);
   const [edited, setEdited] = useState<boolean>(false);
-  const [contract, setContract] = useState<TrackedContract>(trackedContract);
+  const [contract, setContract] = useState<Contract>(trackedContract);
+  // Keep the original contract.
   const initialContract = trackedContract;
 
+  // Reference to the input
   let trackInputRef: any = null;
 
   useEffect(() => {
+    // When adding a new line we want straight awayt to focus the new line
     if (focused && trackInputRef) {
       trackInputRef.focus();
     }
   }, [focused, trackInputRef]);
 
-  // const saved = () => { callParent to update the array view without recharging the page }
-  //
-
   // Handle the UI state for the contrat changing
-  const handleContractChange = (value: TrackedContract): void => {
+  const handleContractChange = (value: Partial<Contract>): void => {
     setContract({ ...contract, ...value });
     setEdited(true);
   };
 
   // reset the state to lose focus on the row
   const loseFocus = (): void => {
     setEdited(false);
-    trackInputRef.blur();
+    setInputFocused(false);
+    console.log('trackinputRef', trackInputRef);
+    if (trackInputRef) {
+      trackInputRef.blur();
+    }
   };
 
   // Save the contract
   const saveClick = (): void => {
     // save contract as it is
```

Comment:
Before saving we would check that the address is valid or not using some utils. Ticket to come about adding a util to check input address based on the chain

Good example from Jure : https://polkadot.js.org/docs/util-crypto/examples/validate-address/

Reply:
This is handled in the backend [here](https://github.com/paralink-network/paralink-node/blob/master/src/models/contract.py#L83).  You should look out for the [error](https://github.com/paralink-network/paralink-node/blob/master/src/api/contracts.py#L61) when calling the `create_contract` endpoint.